uefi/src/table/boot.rs

```diff
@@ -94,7 +94,7 @@ struct BootServicesInternal {
 
     // Protocol handlers
     install_protocol_interface: unsafe extern "efiapi" fn(
-        handle: &mut Option<Handle>,
+        handle: *mut uefi_raw::Handle,
```

Comment:
Why can't we use `Option<&mut uefi_raw::Handle>` here and in similar places? It has the same memory layout but we do have more guarantees in higher-level Rust APIs.

Reply:
We could, but I'm wary of using any references at all in the uefi-raw interface. Broadly, my thinking is that we want uefi-raw to be as close to the C interface as possible, and in particular placing no restrictions on the API that wouldn't also be present for a C user. My understanding is that a Rust reference, even if it's never dereferenced, must always be valid (aligned, non-dangling), and obey the (currently unfinished) [aliasing rules](https://doc.rust-lang.org/nomicon/aliasing.html).